Consuming Multi api Authorization package for stack profile 2017-03-09-profile

bganapa opened the pull request:
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

bganapa commented:
This seems to have gotten other commits from dev after my rebase, looking at it

troydai commented on src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py:
```diff
@@ -1215,6 +1215,9 @@ def _create_role_assignment(cli_ctx, role, assignee, resource_group_name=None, s
     object_id = _resolve_object_id(cli_ctx, assignee) if resolve_assignee else assignee
+    RoleAssignmentCreateParameters = get_sdk(cli_ctx, ResourceType.MGMT_AUTHORIZATION,
+                                             'RoleAssignmentCreateParameters', mod='models',
+                                             operation_group='role_assignments')
```
@yugangw-msft do we have a better way to centralize this piece?

troydai commented on src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py:
```diff
@@ -47,15 +45,40 @@ def list_role_definitions(cmd, name=None, resource_group_name=None, scope=None,
+            setattr(obj, property_name, property_value)
+
+
+def _create_update_role_definition(cmd, role_definition, for_update):
```
This method is too long, please break it down into smaller pieces.

troydai commented on src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py:
```diff
@@ -81,28 +104,48 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
+        raise CLIError('"Permission" configuration is unavailable from the current profile. Please use "az cloud set"'
+                       ' command to use correct one')
+
+    if cmd.supported_api_version(max_api='2015-07-01'):
```
@yugangw-msft is there a better way?

yugangw-msft commented:
@bganapa can you please get CLI green by fixing the authorization RP related tests

yugangw-msft commented on src/command_modules/azure-cli-ams/HISTORY.rst:
```diff
@@ -3,10 +3,18 @@
 +++++
 * BREAKING CHANGE: 'show' commands log error message and fail with exit code of 3 upon a missing resource.
 
+0.1.2
```
back out this change

bganapa replied:
removed

bganapa replied:
removed

yugangw-msft commented on src/command_modules/azure-cli-resource/HISTORY.rst:
```diff
@@ -31,7 +31,6 @@ Release History
 ++++++
 * Minor changes
 
-
```
back out

bganapa replied:
removed

yugangw-msft commented on src/command_modules/azure-cli-role/HISTORY.rst:
```diff
@@ -17,6 +21,7 @@ Release History
 
 2.0.25
 ++++++
+* support stack profile
```
back out

bganapa replied:
removed this line

yugangw-msft commented on src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py:
```diff
@@ -81,28 +104,49 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
+    Permission, RoleDefinition, RoleDefinitionProperties = get_sdk(cmd.cli_ctx, ResourceType.MGMT_AUTHORIZATION,
+                                                                   'Permission', 'RoleDefinition', 'RoleDefinitionProperties', mod='models',
+                                                                   operation_group='role_definitions')
+    if not Permission:
```
please remove this check as `permission` should always be available, even in older api-version

lmazuel commented:
@yugangw-msft @bganapa SDL is already released for a few weeks now, this should work without "privates" folder.

bganapa commented:
@yugangw-msft addressed the comments, the tests are green now. I have run the tests for azurestack profile as well and they are all good.  except for @troydai's ones. Can you please suggest how we can fix his comments?

yugangw-msft commented on src/azure-cli-core/HISTORY.rst:
```diff
@@ -33,10 +37,6 @@ Release History
-2.0.36
-++++++
-* Minor fixes
-
```
why you remove it? My original comment was about removing redundant entries in your PR

bganapa replied:
oops.. i misunderstood your comment, adding it back

yugangw-msft commented on src/command_modules/azure-cli-resource/HISTORY.rst:
```diff
@@ -20,10 +20,6 @@ Release History
 ++++++
 * Support deployment at subscription scope: new operation group `deployment`.
 
-2.0.31
```
Again, why you are removing it?

bganapa replied:
adding it back

yugangw-msft commented on src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py:
```diff
@@ -81,28 +104,50 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
+                                                                   'Permission', 'RoleDefinition',
+                                                                   'RoleDefinitionProperties', mod='models',
+                                                                   operation_group='role_definitions')
+    if not Permission:
```
You don't need to check, as this guy should be available all the time. Please remove this condition

bganapa replied:
removed
